polynote-kernel/src/main/scala/polynote/kernel/lang/LanguageInterpreter.scala

```diff
@@ -17,28 +17,19 @@ import scala.collection.JavaConverters._
   */
 trait LanguageInterpreter[F[_]] {
 
-  // LanguageInterpreter is expected to have a reference to the shared runtime symbol table of a notebook
-  // TODO: eliminate this
-  val symbolTable: RuntimeSymbolTable
-
-  final type Decl = SymbolDecl[F]
 
   def predefCode: Option[String]
 
   /**
     * Run the given code.
     *
-    * @param cell           The identifier string of the cell for the code being run
-    * @param visibleSymbols A list of symbols defined in cells "before" the given code, which are visible to it
-    * @param previousCells  The identifier strings of the cells "before" this code, for stateful language kernels
-    * @param code           The code string to run
+    * @param cellContext  An object describing things about the cell and its dependencies
+    * @param code         The code string to run
     * @return An [[F]] that returns a [[Stream]] which will contain [[Result]] value(s) containing anything that
     *         resulted from running the code.
     */
   def runCode(
-    cell: CellID,
-    visibleSymbols: Seq[Decl],
-    previousCells: Seq[CellID],
+    cellContext: CellContext,
```

Comment:
All these APIs now just take `CellContext`. So for `runCode` it's just the context and the code.